qiskit/providers/basic_provider/basic_simulator.py

```diff
@@ -139,11 +126,109 @@ def __init__(self, configuration=None, provider=None, **fields):
         self._initial_statevector = self.options.get("initial_statevector")
         self._chop_threshold = self.options.get("chop_threashold")
         self._qobj_config = None
-        # TEMP
         self._sample_measure = False
 
+    @property
+    def max_circuits(self) -> None:
+        return None
+
+    @property
+    def target(self) -> Target:
+        if not self._target:
+            if self._configuration is None:
+                self._target = self._build_basic_target()
+            else:
+                self._target = convert_to_target(self._configuration, self._properties)
+        return self._target
+
+    def _build_basic_target(self) -> Target:
+        """Helper method that returns a minimal target with a basis gate set but
+        no coupling map, instruction properties or calibrations.
+
+        Returns:
+            Target: the configured target.
+        """
+        target = Target(
+            description="Basic Target",
+            num_qubits=min(24, self.MAX_QUBITS_MEMORY),
+        )
+        basis_gates = ["h", "u", "p", "u1", "u2", "u3", "rz", "sx", "x", "cx", "id", "unitary"]
+        required = ["measure", "delay", "reset"]
+        parameters = {"unitary": ["matrix"]}
+        inst_mapping = get_standard_gate_name_mapping()
+        for name in basis_gates + required:
+            if name in inst_mapping:
+                instruction = inst_mapping[name]
+            else:
+                instruction = Gate(
+                    name=name,
+                    num_qubits=1 if name == "reset" else 0,
+                    params=list(map(Parameter, parameters.get(name, []))),
+                )
+            target.add_instruction(instruction, properties=None, name=name)
+        return target
+
+    def configuration(self) -> BackendConfiguration:
+        """Return the simulator backend configuration.
+
+        Returns:
+            BackendConfiguration: the configuration for the backend.
```

Comment:
Small comment, but since the type is documented in the signature I think it doesn't need to be here
```suggestion
            The configuration for the backend.
```